sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/RetryUtil.java

```diff
@@ -52,6 +52,27 @@ public static AmqpRetryPolicy getRetryPolicy(AmqpRetryOptions options) {
         }
     }
 
+    /**
+     * Given a {@link Flux} will apply the retry policy to it when the operation times out or throws a transient error.
+     *
+     * @param source The publisher to apply the retry policy to.
+     * @param retryOptions A {@link AmqpRetryOptions}.
+     * @param allowsLongOperation A boolean value indicating whether to allow the {@param source} to run long time
+     *  and not to timeout it. If it's false, a {@link TimeoutException} will be thrown if the {@param source} doesn't
+     *  complete before the {@code getTryTimeout()} of {@param retryOptions}.
+     *
+     * @return A publisher that returns the results of the {@link Flux} if any of the retry attempts are successful.
+     *     Otherwise, propagates a {@link TimeoutException}.
+     */
+    public static <T> Flux<T> withRetry(Flux<T> source, AmqpRetryOptions retryOptions, String timeoutMessage,
```

Comment:
Do we need this overload? I think we are better off just calling  `source.timeout(retryOptions.getTryTimeout())`
The reason I had withRetry was because there was complex logic in constructing the retryWhen operation.